packages/js-runtime/src/runtime/http/URL.ts

```diff
@@ -80,6 +81,48 @@
       throw new Error('Not implemented');
     }
 
+    get href(): string {
+      // TODO: username and password
+      let href = this.protocol + '//' + this.host + this.pathname + this.search + this.hash;
+
+      if (this.protocol === 'file:') {
+        href = href.replace('//', '');
+      }
+
+      return href;
+    }
+
+    set href(href: string) {
+      this.initialize(href);
+    }
+
+    get host(): string {
+      return this.hostname + (this.port ? ':' + this.port : '');
+    }
+
+    set host(host: string) {
+      // eslint-disable-next-line no-useless-escape
+      const result = /^([^:\/?#]*)(?:\:([0-9]+))$/.exec(host);
+      if (result) {
+        const [, hostname, port] = result;
+        this.hostname = hostname;
+        this.port = port;
+      } else {
+        this.hostname = host;
+        this.port = '';
+      }
+    }
+
+    get search(): string {
+      const search = this.searchParams.toString();
+      return search ? '?' + search : '';
+    }
+
+    set search(search: string) {
+      // TODO: Probably not spec compliant, all tested engines keep the same searchParams object
+      this.searchParams = new URLSearchParams(search);
```

Comment:
Instead of creating a new object, we can get and delete all entries and `append()` the new ones by applying the same logic as in `URLSearchParams`'s constructor. Some libs might expect the reference to be the same, which will fail in this case.